### PR TITLE
Add `couplingICubEye` device

### PR DESCRIPTION
**Pull request by martinaxgloria**
This pull request introduces the new `couplingICubEye` coupling handler device.
Need to test it
It implements the IJointCoupling interface:
- https://github.com/robotology/yarp/pull/3026

### CODE REVIEW

**Comment by pattacini** [comment]
We have to get aligned on the name as the Eyes Mk3 do refer to an internal experiment that is obsolete.
Also, Mk2 and Mk4 do have essentially the same mechanism with tiny different kinematic params.

**Comment by Nicogene** [comment]
This PR has been tested with:
- https://github.com/robotology/icub-models/pull/249
- #995  

@pattacini  @traversaro  it is ready to be reviewed/merged

File: src/libraries/icubmod/couplingICubEye/CouplingICubEye.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#include <vector>
+#include <string>
+
+
+YARP_DECLARE_LOG_COMPONENT(COUPLINGICUBEYE)
+
+/**
+ * Coupling law from https://icub-tech-iit.github.io/documentation/icub_kinematics/icub-vergence-version/icub-vergence-version/
+ */
+/** TBD
```
Comment (traversaro) [inline src/libraries/icubmod/couplingICubEye/CouplingICubEye.h:27]: Leftover?

**Review by traversaro — APPROVED** [review]
Minor comments.

**Review by Nicogene — APPROVED** [review]
Tested together!

**Comment by Nicogene** [comment]
> Is the device on by default? If not, could it make sense to enable it by default or add it to the default preset?

Yes it is enabled by default, it does not have external dependencies except YARP

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @Nicogene @martinaxgloria 

Could you please address the conflicts?

**Review by pattacini — APPROVED** [review]
Awaiting the CI before merging.

**Pull request merged**